backend/src/main/java/com/bakdata/conquery/io/mina/PipedJacksonProtocolFilter.java

```diff
@@ -0,0 +1,334 @@
+		final IoBuffer buffer = (IoBuffer) message;
+
+
+		synchronized (session) {
+			do {
+				final Reader asyncReader = getAsyncReader(nextFilter, session, buffer);
+
+				if (asyncReader == null) {
+					// Rare spill-buffer case
+					break;
+				}
+
+				asyncReader.receive(buffer);
+
+				if (asyncReader.hasEnough()) {
+					session.setAttribute(READER_KEY, null);
+				}
+
+			} while (buffer.hasRemaining());
+		}
+	}
+
+	@Nullable
+	private Reader getAsyncReader(NextFilter nextFilter, IoSession session, IoBuffer buffer) throws IOException {
+		final Reader asyncReader = (Reader) session.getAttribute(READER_KEY);
+
+		if (asyncReader != null) {
+			return asyncReader;
+		}
+
+		if (requiresEdgeBuffering(buffer)) {
+
+			final IoBuffer spillBuffer = createSpillBuffer(buffer);
+
+			final Object priorSpill = session.setAttribute(SPILL_BUFFER, spillBuffer);
+			assert priorSpill == null;
+
+			// Buffer contains only partial length we need to combine with the next buffer.
+			return null;
+		}
+
+		final int bufferLength = getBufferLength(session, buffer);
+
+		log.trace("Starting new reader for {}", DataSize.bytes(bufferLength));
+
+		final Reader reader = new Reader(bufferLength, mapper, nextFilter, session, PIPED_IO_BUFFER_SIZE);
+
+		// Enqueue the reader so that it can be bubbled up in order.
+		readersInArrivalOrder.add(reader.getFutureSelf());
+		readerThreads.submit(reader::read);
+
+		session.setAttribute(READER_KEY, reader);
+
+		return reader;
+	}
+
+	@NotNull
+	private static IoBuffer createSpillBuffer(IoBuffer buffer) throws IOException {
+		log.trace("Not enough content in current buffer ({} bytes), spilling.", buffer.remaining());
+
+		final IoBuffer spillBuffer = IoBuffer.allocate(Integer.BYTES, false);
+		spillBuffer.setAutoExpand(false);
+
+		buffer.asInputStream().transferTo(spillBuffer.asOutputStream());
+		return spillBuffer;
+	}
+
+	/**
+	 * Test if we have enough contents to read the incoming buffer's length.
+	 * If not, cumulate the remaining buffer into a new buffer.
+	 * <p>
+	 * Incoming buffers will combine their remaining part of the prefixed-length.
+	 */
+	private boolean requiresEdgeBuffering(IoBuffer buffer) {
+		return buffer.remaining() < Integer.BYTES;
+	}
+
+	/**
+	 * Reads the length of the incoming buffer.
+	 * Incoming messages are always prefixed by their length.
+	 *
+	 * @implNote A single {@link IoBuffer} may contain multiple messages or parts of them.
+	 * In very degenerate cases, the messages ends exactly such that the length of the next messages does not fully fit the current buffer and is spilled over into the next buffer.
+	 * In that case we spill the remaining bytes into a separate buffer and append the incoming few bytes (at most 3!) into a separate buffer to calculate the length.
+	 * This avoids costly copying and reallocation but looks really rather funky.
+	 */
+	private int getBufferLength(IoSession session, IoBuffer buffer) throws IOException {
```

Comment:
Die Bufferlength ist einfach rauszufinden, hier ist die MessageLength gesucht.
```suggestion
	 * @implNote A single {@link IoBuffer} may contain multiple messages or parts of them.
	 * In very degenerate cases, the messages ends exactly such that the length of the next messages does not fully fit the current buffer and is spilled over into the next buffer.
	 * In that case we spill the remaining bytes into a separate buffer and append the incoming few bytes (at most 3!) into a separate buffer to calculate the length.
	 * This avoids costly copying and reallocation but looks really rather funky.
	 */
	private int getMessageLength(IoSession session, IoBuffer buffer) throws IOException {
```